aws-blog-spark-parquet-conversion/README.md

```diff
@@ -0,0 +1,102 @@
+# Converting a large dataset to Parquet
+
+This is a Spark script that can read data from a Hive table and convert the dataset to the Parquet format. We will be using a combination of Spark and Python native threads to convert a 1 TB CSV dataset to Parquet in batches. Our sample dataset is 1 year of ELB log data in S3 available as a Hive External Table, and we will be converting this dataset to Parquet in 3 batches, each batch comprising of 4 months of data. As we are reading the data from a Hive Table, we can use this script to convert CSV, TSV, JSON or any Hive supported format to Parquet or ORC files.
+
+## Prerequisites
+- Amazon Web Services account
+- [AWS Command Line Interface (CLI)]
+
+#### Launching an AWS EMR Spark cluster
+
+You can either launch a Spark cluster from the AWS console or using the AWS CLI as below. For this purpose, we will be selecting a cluster of 2 r3.8xlarge nodes. This will ensure that the memory avalable to the cluster will be greater than 333 GB to accomodate 4 months of data in each run. We will also be adding EBS storage volumes to ensure we can accomodate the intermediate and output data in HDFS. Please ensure that the EMR Master Security Group allows you SSH access.
+
+```
+aws emr create-cluster --applications Name=Hadoop Name=Hive Name=Spark Name=Ganglia Name=Tez \
+--ec2-attributes '{  
+  "KeyName":"<<KEY NAME>>",
+  "InstanceProfile":"EMR_EC2_DefaultRole",
+  "SubnetId":"<<SUBNET ID>>",
+  "EmrManagedSlaveSecurityGroup":"<<SECURITY GROUP ID>>",
+  "EmrManagedMasterSecurityGroup":"<<SECURITY GROUP ID>>"
+}'  \
+--service-role EMR_DefaultRole --release-label emr-5.2.0 \
+--name 'Spark Cluster' --instance-groups '[  
+  {  
+    "InstanceCount":1,
+    "InstanceGroupType":"MASTER",
+    "InstanceType":"m3.xlarge",
+    "Name":"Master instance group - 1"
+  },
+  {  
+    "InstanceCount":2,
+    "EbsConfiguration":{  
+      "EbsBlockDeviceConfigs":[  
+        {  
+          "VolumeSpecification":{  
+            "SizeInGB":840,
+            "VolumeType":"gp2"
+          },
+          "VolumesPerInstance":1
+        }
+      ],
+      "EbsOptimized":false
+    },
+    "InstanceGroupType":"CORE",
+    "InstanceType":"r3.8xlarge",
+    "Name":"Core instance group - 2"
+  }
+]'  \
+--region us-east-1
+```
+
+##### Creating SSH Tunnel to the EMR Master Node
```

Comment:
if you're not explaining how to use the web interfaces (Resource Manager / Ganglia) this step can be removed.